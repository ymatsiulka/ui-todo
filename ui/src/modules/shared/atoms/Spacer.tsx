import React from 'react';
import { type SpacerProps } from 'types/frontend';

const Spacer: React.FC<SpacerProps> = ({ ...props }) => {
  const { top, bottom, left, right } = { ...props };
  return <div {...props} style={{ marginTop: top, marginBottom: bottom, marginLeft: left, marginRight: right }} />;
};

export default Spacer;
