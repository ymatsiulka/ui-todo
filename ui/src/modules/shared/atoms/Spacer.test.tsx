import React from 'react';
import { render } from '@testing-library/react';
import Spacer from './Spacer';

test('Spacer should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(
    <Spacer className="spacer" top={1} bottom={1} left={1} right={2}>
      Some text
    </Spacer>,
  );

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
