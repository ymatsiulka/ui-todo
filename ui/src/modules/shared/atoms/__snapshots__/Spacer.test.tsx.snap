// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render properly 1`] = `
<DocumentFragment>
  <div
    bottom="1"
    class="spacer"
    left="1"
    right="2"
    style="margin: 1px 2px 1px 1px;"
    top="1"
  >
    Some text
  </div>
</DocumentFragment>
`;
